### ISSUE TITLE
Moving the cursor above/below the viewport doesn't scroll the window to the cursor's new position

### ISSUE THREAD

**Issue by aaronhall** [WuTheFWasThat/vimflowy#246]
Moving the cursor to a position that isn't visible in the window (like using `G` or `gg` on long lists) doesn't scroll the window to that position. The window doesn't scroll to the cursor until the next action/movement.

**Comment by WuTheFWasThat** [comment]
The window scrolls properly for me - I can't reproduce the issue.  What browser are you using?  

After doing `gg` so that the cursor is off-screen, could you try this in javascript console:

```
function scrollDiv($elem, amount) {
  return $elem.scrollTop($elem.scrollTop() + amount);
}

function scrollIntoView(el, $within, margin) {
  const elemTop = el.getBoundingClientRect().top;
  const elemBottom = el.getBoundingClientRect().bottom;

  const top_margin = margin;
  const bottom_margin = margin + $('#bottom-bar').height();

  if (elemTop < top_margin) {
    // scroll up
    return scrollDiv($within, elemTop - top_margin);
  } else if (elemBottom > window.innerHeight - bottom_margin) {
    // scroll down
    return scrollDiv($within, elemBottom - window.innerHeight + bottom_margin);
  }
}

const $mainDiv = $('#view');
const cursorDiv = $('.cursor', $mainDiv)[0];
if (cursorDiv) {
  utils.scrollIntoView(cursorDiv, $mainDiv, 50);
}
```

It should be equivalent to what is happening in the function `renderMain` in `src/assets/ts/app.tsx`.  If it doesn't work, it's hopefully possible to debug why.

**Comment by WuTheFWasThat** [comment]
nevermind, it turns out I broke this earlier today... fixed in 6b3df4adc035c059cd6ca552e4826ef124d6b1df.  I'm guessing you were running locally on a recent commit?  thanks for reporting the bug, and feel free to re-open if it's not in fact fixed

**Comment by aaronhall** [comment]
> I'm guessing you were running locally on a recent commit?

Yep, within the last couple of hours. Works for me @ 6b3df4a - thanks
